packages/cli/src/commonlib/m365Login.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+"use strict";
+
+import {
+  err,
+  FxError,
+  M365TokenProvider,
+  ok,
+  Result,
+  TokenRequest,
+  UserError,
+} from "@microsoft/teamsfx-api";
+import { LogLevel } from "@azure/msal-node";
+import { CodeFlowLogin, ConvertTokenToJson, ErrorMessage } from "./codeFlowLogin";
+import CLILogProvider from "./log";
+import { CryptoCachePlugin } from "./cacheAccess";
+import { m365CacheName, signedIn, signedOut } from "./common/constant";
+import { LoginStatus } from "./common/login";
+import { BasicLogin } from "@microsoft/teamsfx-core";
+import M365TokenProviderUserPassword from "./m365LoginUserPassword";
+
+const SERVER_PORT = 0;
+
+const cachePlugin = new CryptoCachePlugin(m365CacheName);
+
+const config = {
+  auth: {
+    clientId: "7ea7c24c-b1f6-4a20-9d11-9ae12e9e7ac0",
+    authority: "https://login.microsoftonline.com/common",
+  },
+  system: {
+    loggerOptions: {
+      loggerCallback(loglevel: any, message: any, containsPii: any) {
+        if (this.logLevel <= LogLevel.Error) {
+          CLILogProvider.log(4 - loglevel, message);
+        }
+      },
+      piiLoggingEnabled: false,
+      logLevel: LogLevel.Error,
+    },
+  },
+  cache: {
+    cachePlugin,
+  },
+};
+
+export class M365Login extends BasicLogin implements M365TokenProvider {
+  private static instance: M365Login;
+  private static codeFlowInstance: CodeFlowLogin;
+
+  private constructor() {
+    super();
+    M365Login.codeFlowInstance = new CodeFlowLogin([], config, SERVER_PORT, "m365");
```

Comment:
Does this mean we deprecate the old token cache file?

Reply:
Oh, I see, it's controlled by cache name, not account name.

Reply:
But what's the side effect if changing account name from `appStudio` to `m365`?

Reply:
My bad, this should be "appStudio". Because `CodeFlowLogin` use this name to manipulate cache file. If we change from `appStudio` to `m365`, the user need to login again if he has already signed in.